Hoxtygen/next-fullstack
# Feat: implement user listing delete (HOM-73)

## What does this PR do?
Implements a feature that enables user to delete listings

## Description of the tasks to be completed
* Add dialog component to confirm deletion
* Integrate with listing deletion endpoint

## How should this be manually tested?
* Clone the app
* Open a terminal on your PC and run `git clone <repo_url>`
* `cd` into the project directory and run `yarn install`.
* After package installation run `yarn dev`
* Open a tab on your browser and navigate to `http://localhost:3000/auth/signup` to sign up  or `http://localhost:3000/auth/login` to log in if you already have an account.
* Upload a listing
* Navigate to the listings page and click on the uploaded listing
* On the listings detail page click on the delete button.
* Click on the confirm button on the dialog to confirm the deletion.

## What are the relevant Jira stories
[HOM-73](https://wasiu-dev.atlassian.net/jira/software/projects/HOM/boards/2?selectedIssue=HOM-73)

[HOM-73]: https://wasiu-dev.atlassian.net/browse/HOM-73?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
